src/lib/mina_base/transaction_union_payload.ml

```diff
@@ -164,15 +138,20 @@ module Body = struct
       { tag = Tag.unpacked_of_t tag
       ; source_pk = Public_key.Compressed.var_of_t source_pk
       ; receiver_pk = Public_key.Compressed.var_of_t receiver_pk
-      ; token_id = Token_id.var_of_t token_id
+      ; token_id = Tid.Checked.constant token_id
       ; amount = Currency.Amount.var_of_t amount
       ; token_locked = Boolean.var_of_value token_locked
       }
 
     let to_input_legacy
         { tag; source_pk; receiver_pk; token_id; amount; token_locked } =
-      let%map token_id = Token_id.Checked.to_input_legacy token_id
-      and amount = Currency.Amount.var_to_input_legacy amount in
+      let%map token_id =
+        Mina_numbers.Token_id.(Checked.to_input_legacy (var_of_t default))
```

Comment:
It seems all these `to_input_legacy` methods are mainly used by rosetta/lib and client_sdk? Would be nice to get rid of them

Reply:
They're used for the signatures of old transactions, so that we don't break existing signing tools at the hard fork. Sadly, we need to keep them.

Reply:
It would be nice to hard-code the value here, rather than keeping `Mina_numbers.Token_id` around.